src/webview/components/memory-table.tsx

```diff
@@ -151,18 +170,50 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
         };
     }
 
+    componentDidMount(): void {
+        this.resizeObserver = new ResizeObserver(entries => {
+            // The size changed - we could have too few rows visible to enable a scrollbar
+            if (entries.length > 0 && this.props.scrollingBehavior === 'Auto-Append') {
+                this.assertSufficientVisibleRowsForScrollbar();
+            }
+        });
+
+        const element = this.datatableRef.current?.getElement();
+        if (element) {
+            this.resizeObserver.observe(element);
+        }
+    }
+
     componentDidUpdate(prevProps: Readonly<MemoryTableProps>): void {
-        const hasMemoryChanged = prevProps.memory?.address !== this.props.memory?.address || prevProps.offset !== this.props.offset || prevProps.count !== this.props.count;
-        const hasOptionsChanged = prevProps.wordsPerGroup !== this.props.wordsPerGroup || prevProps.groupsPerRow !== this.props.groupsPerRow;
+        if (MemoryState.hasMemory(prevProps) && MemoryState.hasMemory(this.props)) {
```

Comment:
Optional style nitpick: I'd invert the condition and return, to avoid nesting the entire remainder of the method.